flang/lib/Optimizer/Builder/Runtime/Command.cpp

```diff
@@ -19,3 +20,46 @@ mlir::Value fir::runtime::genCommandArgumentCount(fir::FirOpBuilder &builder,
       fir::runtime::getRuntimeFunc<mkRTKey(ArgumentCount)>(loc, builder);
   return builder.create<fir::CallOp>(loc, argumentCountFunc).getResult(0);
 }
+
+void fir::runtime::genGetCommandArgument(fir::FirOpBuilder &builder,
+                                         mlir::Location loc, mlir::Value number,
+                                         mlir::Value value, mlir::Value length,
+                                         mlir::Value status,
+                                         mlir::Value errmsg) {
+  auto argumentValueFunc =
+      fir::runtime::getRuntimeFunc<mkRTKey(ArgumentValue)>(loc, builder);
+  auto argumentLengthFunc =
+      fir::runtime::getRuntimeFunc<mkRTKey(ArgumentLength)>(loc, builder);
+
+  auto isPresent = [&](mlir::Value val) -> bool {
+    auto definingOp = val.getDefiningOp();
+    if (auto cst = mlir::dyn_cast<fir::AbsentOp>(definingOp))
+      return false;
+    return true;
+  };
+
+  if (isPresent(value) || status || isPresent(errmsg)) {
```

Comment:
All arguments are optional, but you only apply `isPresent` to `value` and `errmsg`. Why?

Reply:
So value, status and errmsg are all "INTENT (OUT)". value and errmsg are parameters required to call the intrinsic function `ArgumentValue`. However status is a variable that we "set" from the result of `ArgumentValue`. For these reasons, when we process the parameters, value/errmsg "must" always have a value whereas status doesn't necessarily need one. In the case where value/errmsg don't have a value, we give it an "absent" value (similar to a nullptr). For this reason, when checking if a value exists, we can't just use "mlir::Value::getImpl()" (which is that the operator bool calls) for value/errmsg, as they have a impl from absent, whereas status does not.